docs/src/inverse_problems/petab_ode_param_fitting.md

```diff
@@ -0,0 +1,494 @@
+```@example petab1
+petab_problem = PEtabODEProblem(petab_model, 
+                                ode_solver=ODESolver(Rodas5P(), abstol=1e-8, reltol=1e-8), 
+                                gradient_method=:ForwardDiff, 
+                                hessian_method=:ForwardDiff)
+nothing # hide
+```
+where we simulate our ODE model using the `Rodas5p` method (with absolute and relative tolerance both equal `1e-8`) and use [forward automatic differentiation](https://github.com/JuliaDiff/ForwardDiff.jl) for both gradient and hessian computation. More details on available ODE solver options can be found in [the PEtab.jl documentation](https://sebapersson.github.io/PEtab.jl/dev/API_choosen/#PEtab.ODESolver).
+
+## [Additional features: Optimisation](@id petab_optimisation)
+
+### [Optimisation methods and options](@id petab_optimisation_optimisers)
+For our examples, we have used the `Optim.IPNewton` optimisation method. PEtab.jl supports [several additional optimisation methods](https://sebapersson.github.io/PEtab.jl/dev/Avaible_optimisers/#options_optimisers). Furthermore, `calibrate_model`'s `options` argument permits the customisation of the options for any used optimiser. E.g. to designate the maximum number of iterations of the `Optim.IPNewton` method we would use:
+```@example petab1
+res = calibrate_model(petab_problem, p0, IPNewton(); options=Optim.Options(iterations = 10000))
+nothing # hide
+```
+
+Please read the [PEtab.jl documentation](https://sebapersson.github.io/PEtab.jl/dev/Avaible_optimisers/#options_optimisers) to learn how to customise the various optimisers' properties. 
+
+### [Optimisation path recording](@id petab_optimisation_path_recording)
+
+To record all the parameter sets evaluated (and their objective values) during the optimisation procedure, provide the `save_trace=true` argument to `calibrate_model` (or `calibrate_model_multistart`). This is required for the various [optimisation evaluation plots](@ref petab_plotting) provided by PEtab.jl. If desired, this information can be accessed in the calibration output's `.xtrace` and `.ftrace` fields.
+
+
+## Objective function extraction
+While PEtab.jl provides various tools for analysing the objective function generated by `PEtabODEProblem`, it is also possible to extract this function for customised analysis. Given a `PEtabODEProblem`
+```@example petab1
+petab_problem = PEtabODEProblem(petab_model)
+nothing # hide
+```
+We can find the:
+1. Objective function in the `petab_problem.compute_cost`. It takes a single argument (`p`) and returns the objective value.
+2. Gradient in the `petab_problem.compute_gradient!` field. It takes two arguments (`g` and `p`) with the updated gradient values being written to `g`.
+3. Hessian in the `petab_problem.compute_hessian` field. It takes two arguments (`H` and `p`) with the updated hessian values being written to `H`.
```

Comment:
```suggestion
3. Hessian as `petab_problem.compute_hessian` field. It takes two arguments (`H` and `p`) with the updated hessian values being written to `H`.
```

Reply:
Why use "as" for the Hessian, but not for the Gradient and the objective function?